### DISCUSSION TITLE
Collection of All `preact/signals` powered libraries

### DISCUSSION THREAD

**Discussion by XantreDev** [preactjs/signals#397] [General]
Hello everyone! 🎉

In an effort to consolidate resources and help the community navigate the growing ecosystem, i want to create a central list for all libraries that are powered by `preact/signals`.

If you've developed or come across a library that works seamlessly with `preact/signals`, please add it to this discussion. Your contributions will help others find the right tools and resources to build amazing reactive applications using Preact Signals.

### Guidelines:

1. **One Library Per Comment**: To keep things organized, please list one library per comment. This will make it easier for others to discuss, upvote, or ask questions about a specific library.
2. **Library Details**: Include the library's name, a brief description, its main features, and a link to its repository or npm page.
3. **Check Before Adding**: Before adding a new library to the list, please check the existing comments to avoid duplicates.

### Template:

```markdown
**Library Name**: [Your Library Name Here]
**Description**: A brief description of what the library does.
**Main Features**:
- Feature 1
- Feature 2
- ...
**Link**: [GitHub/NPM link here]
```

Note: If there are any additional points or guidelines you want to add or modify, feel free to do so

**Comment by XantreDev** [comment]
**Library Name**: `@preact-signals/query`

**Description**: An integration of the renowned `@tanstack/query-core` with Preact Signals, tailored specifically to enable truly reactive data fetching and state synchronization in React/Preact apps.

**Main Features**:
- Seamless Integration: Drop-in replacement for `@tanstack/react-query` with shared tests and source code.
- Reactive Hooks: Introduces signal-compatible query hooks like `useQuery$`, `useMutation$`, `useQueryClient$`, and `useIsFetching$`.
- Suspense and Error Handling: Supports on-demand suspense and error boundary triggering based on reactive signal consumption.
- Flat-store Returns: Optimized for Preact Signals, query hooks return a flat-store, ensuring efficient and accurate reactivity.
- Enhanced Mutation Support: `useMutation$` hook with added features and improvements over its `react-query` counterpart.

**Link**: [npm](https://www.npmjs.com/package/@preact-signals/query), [github](https://github.com/XantreGodlike/preact-signals/tree/main/packages/query)

**Comment by XantreDev** [comment]
**Library Name**: `@preact-signals/utils`

**Description**: A lot of utils, that will be really handy for preact signals usage with React/Preact or without.

**Main Features**:
- **Adaptive Code Splitting**: Split code based on your preference, not obligations.
- **Reactive Utilities**: Library-agnostic utils, components, hooks, and more to supercharge reactivity.
- **Deep Reactivity**: Intuitive deep-state management akin to Vue.
- **Reactive Components**: Components like Show, Switch, and For to simplify JSX logic.
- **Hooks**: Signal-specific hooks for reactivity in components.
- **High Order Components (HOCs)**: Enhance components by injecting signals or reactive props.


**Link**: [npm](https://www.npmjs.com/package/@preact-signals/utils), [github](https://github.com/XantreGodlike/preact-signals/tree/main/packages/utils)

**Comment by JonAbrams** [comment]
**Library Name**: signals-react-safe
**Description**: A wrapper around `@preact/signals-core` to make Signals useable within any React project, without patching React's internals. Use as a "safer" alternative to `@preact/signals-react`.
**Main Features**:
- Provides the same hooks provided by `signals-react` like `useSignal`.
- Provides new hooks to read signal values in components: `useSignalValue`, `useSignalAndValue`, and `useComputeValue`.
- Mutate signals the way you're used to.
- Render signals directly in JSX/TSX like you're used to, wtih the same performance gains.
**Link**: [github](https://github.com/JonAbrams/signals-react-safe) [npm](https://www.npmjs.com/package/signals-react-safe)
